src/module.c

```diff
@@ -6604,7 +6604,7 @@ void *moduleLoadString(RedisModuleIO *io, int plain, size_t *lenptr) {
         if (opcode != RDB_MODULE_OPCODE_STRING) goto loaderr;
     }
     void *s = rdbGenericLoadStringObject(io->rio,
-              plain ? RDB_LOAD_PLAIN : RDB_LOAD_NONE, lenptr);
+              plain ? RDB_LOAD_PLAIN : RDB_LOAD_ENC, lenptr);
```

Comment:
looks like this change can also cause them to get encoded as OBJ_ENCODING_INT, which means RM_StringPtrLen won't work.

Reply:
I missed that. I pushed a fix.